x-pack/plugins/enterprise_search/server/lib/check_access.ts

```diff
@@ -43,7 +43,7 @@ export const checkAccess = async ({
   request,
   log,
 }: CheckAccess): Promise<ProductAccess> => {
-  const isRbacEnabled = security?.authz.mode.useRbacForRequest(request) ?? false;
+  const isRbacEnabled = security.authz.mode.useRbacForRequest(request) ?? false;
 
   // We can only retrieve the active space when either:
   // 1) security is enabled, and the request has already been authenticated
```

Comment:
@legrego I'm not as familiar with the various security/spaces check in this file as I should be. Now that both security **and** spaces plugins are required by Enterprise Search, is there any extra checks we can now clean up / skip in this file as a result?

Reply:
Sorry, also want to note for this - I'm trying to get this PR in quickly to meet the alpha BC1 FF, so I may not have time for a cleanup revision right now, but would definitely love a follow-up cleanup PR!

Reply:
I think the only thing you can clean up here is the conditional checks on whether or not the `spaces` plugin is defined.

Even though the Kibana `security` plugin is required, we still have to check whether or not stack security features are enabled in ES, and that is done via the call to `security.authz.mode.useRbacForRequest`

Reply:
Ahhhh okay. I think I actually need to modify `check_access.ts` in that case - in 8.0 the underlying Enterprise Search service will no longer work without x-pack security, so we should do an early return to manually disable the plugin if `isRbacEnabled` is false - does that sound right @ioanatia?

Reply:
PR with access changes: https://github.com/elastic/kibana/pull/106307/commits/9bcea085210c1699aa6decdcaf2a3aa1bdb9f0a4

Reply:
Would super appreciate your eyes on that commit @legrego, but no worries if you're done for the day! As I mentioned, happy to open a follow-up PR 💃